### ISSUE TITLE
[no-constant-condition] fails when both sides are explicitly coerced

### ISSUE THREAD

**Issue by alitaheri** [buzinas/tslint-eslint-rules#130]
### Bug Report
- **TSLint version**: `3.15.1`
- **TSLint ESLint Rules version**: `2.1.0`
- **TypeScript version**: `nightly 2.1.0-dev.20161029`
- **Running TSLint via**: CLI
#### TypeScript code being linted

``` ts
const a = '1';
const b = '2';

// This is valid code, no constant condition here.
if (+a > +b) {
 console.log(a);
}
```

with `tslint.json` configuration:

``` json
{
  "no-constant-condition": true,
}
```
#### Actual behavior

`unexpected constant condition`
#### Expected behavior

No warnings.

**Comment by jmlopez-rod** [comment]
I think this is beyond the current scope of the rule. See the eslint documentation for the rule: http://eslint.org/docs/rules/no-constant-condition.

The rule can't do analysis to determine that you have a constant expression. I think the rule was made to prevent debugging code from making it in production. Currently a constant expression 
can be a:

- String Literal
- Function Expression
- Object Expression
- Array Expression

Would you consider the next code to be a candidate for the rule to complain?

```
const a = { value: true };

if (a.value) {
 console.log(a);
}
```

**Comment by alitaheri** [comment]
@jmlopez-rod I'm sorry, I actually mixed up the actual and expected behaviors 😅 

I mean, it shouldn't warn me for that, but it does. This is not a feature request it's a bug report :grin:

**Comment by jmlopez-rod** [comment]
@alitaheri , thanks, I have submitted a PR with a fix.
